validator/remote/src/main/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapter.java

```diff
@@ -188,7 +189,7 @@ private boolean findReadyFailoverAndSwitch() {
         failoverBeaconNodeApis.stream()
             .filter(beaconNodeReadinessManager::isReady)
             .filter(failover -> !currentEventStreamHasSameEndpoint(failover))
-            .findFirst();
+            .max(Comparator.comparing(beaconNodeReadinessManager::getReadinessStatus));
```

Comment:
We are comparing the ordinal of the ReadinessStatus enum but shouldn't we consider the `weight`? It's technically the same thing, so can either use one or the other in my opinion. If we use the ordinal, the `isReady()` check can become simply `!= NOT_READY` and get rid of the weight

Reply:
Good point, however I'd rather use the `weight` to make the concept clear and avoid any confusion when maintaining it in the future... Not a fan of using enum ordinals since they're made for sophisticated data structures

Reply:
Yeah I agree. The change looks good now. I would just rename `getReadinessStatus` to `getReadinessStatusWeight`

Reply:
Done